tests/kun_runtime/test_collab_policy.py

```
@@ -0,0 +1,154 @@
+
+        async def await_user_input(payload):
+            return {"status": "submitted", "answers": {"choice_id": "cancel", "choice_label": "取消"}}
+
+        host = self._host()
+        call = ToolCallLike(call_id="c1", tool_name="write_file", arguments={"path": "/tmp/ws/a.txt", "content": "x"})
+        result = asyncio.run(host.execute(call, self._ctx(mode="manual", await_user_input=await_user_input)))
+        assert result.item["status"] == "cancelled"
+        assert result.item["isError"] is True
+
+    def test_no_channel_falls_back_to_execute(self):
+        """无 await_user_input 通道（headless）时 collab gate 不阻塞，走正常执行。"""
```

Ruff (0.16.1)

[warning] Docstring contains ambiguous `（` (FULLWIDTH LEFT PARENTHESIS). Did you mean `(` (LEFT PARENTHESIS)?

(RUF002)